src/dmd/cparse.d

```diff
@@ -3382,8 +3382,21 @@ final class CParser(AST) : Parser!AST
                     break;
 
                 case TOK.leftParenthesis:
-                    if (!skipParens(t, &t))
+                    if (!skipParens(t, &t)) {
                         return false;
+                    } else {
+                        /*
+                            Fix issue 22267: If the parser encounters the following
+                                `identifier variableName = (expression);`
+                            the initializer is not identified as such since the parentheses
+                            confuses the parser.
+
+                            As such, we just tell the rest of this function to assume
+                            we are in An AssignExpression. This function is ad-hoc as-is so
+                            fuggedaboutit.
+                        */
+                        any = true;
+                    }
```

Comment:
```suggestion
                    if (!skipParens(t, &t))
                        return false;

                    /* https://issues.dlang.org/show_bug.cgi?id=22267
                       If the parser encounters `identifier variableName = (expression);`
                       the initializer is not identified as such since the parentheses
                       were ignored.
                      */
                      any = true;
```